### PR TITLE
fix: update libdatadog to v14, which fixes crashtracking

### PR DESCRIPTION
**Pull request by sanchda**
Prior to the v14.0.0 release of libdatadog, crashtracking could potentially result in the generation of zombie processes.  This was especially evident in environments where short-lived forked workers were used.

This patch
* updates libdatadog to use the new version of crashtracking
* fixes the calling APIs (in particular, this splits the `altstack` setting into a `use_altstack` and a `create_altstack` configuration, which reflects the changes to the inner API and is also useful for diagnostics--most of the code changes come from this pointless interfacing noise)
* adds tests for the two known pathological cases this fix is intended to address
* eliminates an old shim signal handler we used to squelch an error erroneously printed to stderr

I'm marking this as a chore because customers have not really encountered this issue and it's not tied to the behavior of a user-facing component.  This is precautionary.

## Checklist
- [X] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: tests/internal/crashtracker/test_crashtracker.py [hunk 1]
```diff
@@ -579,3 +579,144 @@ def test_crashtracker_user_tags_core():
...
+    # The child process emulates a worker fork in the sense that we spawn a number of them in the parent and then
+    # do a timed `waitpid()` anticipating ECHILD until they all exit.
+    children = []
+    for _ in range(5):
+        pid = os.fork()
+        if pid == 0:
+            if not utils.start_crashtracker(port):
+                with open(err_file, "a") as f:
+                    f.write("X")
+                    exit(-1)
```
[the remark this answers is not in the document]
Comment (sanchda) [reply tests/internal/crashtracker/test_crashtracker.py:621]: Fine

**Review by P403n1x87 — APPROVED** [review]

File: tests/internal/crashtracker/utils.py [hunk 2]
```diff
@@ -82,3 +81,61 @@ def read_files(files):
...
+# A class that wraps start_crashtracker and maintains its own logfiles
+class CrashtrackerWrapper:
+    _seed = 0
+
+    def __init__(self, port: int, base_name=""):
+        if CrashtrackerWrapper._seed == 0:
+            CrashtrackerWrapper._seed = random.randint(0, 999999)
+
+        self.port = port
+        self.stdout = "stdout." + base_name + ".%d.log" % CrashtrackerWrapper._seed
```
[the remark this answers is not in the document]
Comment (sanchda) [reply tests/internal/crashtracker/utils.py]: ok

File: tests/internal/crashtracker/utils.py [hunk 3]
```diff
@@ -82,3 +81,61 @@
...
+        self.stderr = "stderr." + base_name + ".%d.log" % CrashtrackerWrapper._seed
```
[the remark this answers is not in the document]
Comment (sanchda) [reply tests/internal/crashtracker/utils.py]: ok

**Review by brettlangdon — APPROVED** [review]

File: ddtrace/internal/datadog/profiling/dd_wrapper/include/constants.hpp [hunk 4]
```diff
@@ -11,8 +11,8 @@ constexpr unsigned int g_default_max_nframes = 64;
 // their stacks may be silently truncated, which is unfortunate.
 constexpr unsigned int g_backend_max_nframes = 512;
 
-// Maximum amount of time, in seconds, to wait for crashtracker send operations
-constexpr uint64_t g_crashtracker_timeout_secs = 5;
+// Maximum amount of time, in seconds, to wait for crashtracker signal handler
```
[the remark this answers is not in the document]
Comment (sanchda) [reply ddtrace/internal/datadog/profiling/dd_wrapper/include/constants.hpp:14]: Oops, missed that one--thanks!

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**